docs/data/data-grid/overlays/LoadingOverlayVariants.js

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import { DataGrid } from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+export default function LoadingOverlayVariants() {
+  const [withRows, setWithRows] = React.useState(false);
+  const toggleRows = () => setWithRows((prevwithRows) => !prevwithRows);
+
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 5,
+    maxColumns: 9,
+  });
+
+  return (
+    <Box sx={{ width: '100%' }}>
+      <Stack direction="row" spacing={1} sx={{ mb: 1 }}>
+        <Button size="small" onClick={toggleRows}>
+          {withRows ? 'Remove rows' : 'Add rows'}
+        </Button>
```

Comment:
How about adding two `Select`s here for `variant` and `noRowsVariant`?
That would be the ultimate demo for the loading overlay.

Reply:
Updated the demo based on this suggestion, works well I think https://deploy-preview-13293--material-ui-x.netlify.app/x/react-data-grid/overlays/#variants